src/rtextures.c

```diff
@@ -1630,6 +1630,241 @@ Image ImageTextEx(Font font, const char *text, float fontSize, float spacing, Co
     return imText;
 }
 
+Image ImageFromChannel(Image image, int selected_channel, float threshold)
+{
+    Image result = { 0 };
+
+    // Security check to avoid program crash
+    if ((image.data == NULL) || (image.width == 0) || (image.height == 0))
+        return result;
+
+    // Check thresholds limits
+    if (threshold < 0.0f)
+    {
+        TRACELOG(LOG_WARNING, "Threshold is from 0 to 1, changing to 0.");
+        threshold = 0.0f;
+    }
+    if (threshold > 1.0f)
+    {
+        TRACELOG(LOG_WARNING, "Threshold is from 0 to 1, changing to 1.");
+        threshold = 1.0f;
+    }
+
+    // Check selected channel
+    if (image.format == PIXELFORMAT_UNCOMPRESSED_GRAYSCALE
+            || image.format == PIXELFORMAT_UNCOMPRESSED_R32
+            || image.format == PIXELFORMAT_UNCOMPRESSED_R16
+            )
+    {
+        if (selected_channel > 0)
```

Comment:
Please, could you review [raylib naming conventions](https://github.com/raysan5/raylib/blob/master/CONVENTIONS.md)